jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/gcprules/vminstance/IPFowardingRule.java

```diff
@@ -89,7 +89,7 @@ private boolean checkIPForwardRule(String vmEsURL, Map<String, Object> mustFilte
             }
             if (!vmInstanceObject.get(PacmanRuleConstants.TAGS).getAsJsonObject().isJsonNull()) {
                 JsonObject tags = vmInstanceObject.get(PacmanRuleConstants.TAGS).getAsJsonObject();
-                if (!tags.get(PacmanRuleConstants.GKE_CLUSTER_NAME).isJsonNull() && tags.get(PacmanRuleConstants.GKE_CLUSTER_NAME).getAsString().isEmpty()) {
+                if (tags.has(PacmanRuleConstants.GKE_CLUSTER_NAME) && !tags.get(PacmanRuleConstants.GKE_CLUSTER_NAME).isJsonNull() && tags.get(PacmanRuleConstants.GKE_CLUSTER_NAME).getAsString().isEmpty()) {
```

Comment:
Why are so many checks needed. A better solution would be to handle the issue at the source to guarantee consistency. 

@santhosh-challa @AishwaryaKulkarni97 if the data issue is at the collector level, let's fix the collector(s).

Remember the separation of concerns (who is responsible for what).

Reply:
I have another question - why we checking tags for IP forwarding? Are those tags mandatory and cannot be removed? If it's not true then from my understanding customer can remove (or even never have) them but still have IP Forwarding on. ANd we will be firing violations for nothing. Is it correct or not?

Reply:
Its not a collector issue because tags may or may not have any values

Reply:
@mkinchin-paladin instance created by GKE should not be considered for this policy